app/src/main/java/org/mozilla/fenix/home/collections/TabInCollection.kt

```diff
@@ -0,0 +1,262 @@
+                iconPainter = painterResource(drawable.ic_close),
+                iconDescription = stringResource(string.remove_tab_from_collection),
+                onIconClick = { onRemove(false) },
+            )
+        }
+    }
+}
+
+/**
+ * Composable used to display the background of a [Tab] shown in collections that is being swiped left or right.
+ *
+ * @param dismissDirection [DismissDirection] of the tab being swiped depending on which this composable
+ * will also indicate the swipe direction by placing a warning icon at the start of the swipe gesture.
+ * If `null` the warning icon will be shown at both ends.
+ * @param isLastInCollection Whether the tab is to be shown between others or as the last one in collection.
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun DismissedTabBackground(
+    dismissDirection: DismissDirection?,
+    isLastInCollection: Boolean,
+) {
+    Card(
+        modifier = Modifier.fillMaxSize(),
+        backgroundColor = FirefoxTheme.colors.layer3,
+        shape = if (isLastInCollection) bottomTabShape else middleTabShape,
+        elevation = 0.dp,
+    ) {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == StartToEnd) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+
+            Icon(
+                painter = painterResource(drawable.ic_delete),
+                contentDescription = null,
+                modifier = Modifier
+                    .padding(horizontal = 32.dp)
+                    // Only show the delete icon for where the swipe starts.
+                    .alpha(
+                        if (dismissDirection == EndToStart) 1f else 0f
+                    ),
+                tint = FirefoxTheme.colors.iconWarning,
+            )
+        }
+    }
+}
+
+/**
+ * Clips the Composable this applies to such that it cannot draw content / shadows outside it's top bound.
+ */
+private fun Modifier.clipTop() = this.then(
+    Modifier.drawWithContent {
+        val paddingPx = Constraints.Infinity.toFloat()
+        clipRect(
+            left = 0f - paddingPx,
+            top = 0f,
+            right = size.width + paddingPx,
+            bottom = size.height + paddingPx
+        ) {
+            this@drawWithContent.drawContent()
+        }
+    }
+)
+
+/**
+ * Get a friendlier short url for [url].
+ *
+ * @param url Full url to be shortened.
+ *
+ * @see toShortUrl
+ */
+@Composable
+private fun shortenUrl(url: String): String {
```

Comment:
We landed an extension function for `TabSessionState` that does the same thing in https://github.com/mozilla-mobile/fenix/pull/25242. It's kinda unfortunate, but maybe we should've actually add a `String` extension function instead of `TabSessionState`

Reply:
Not sure I have to do anything about this now.